### PR TITLE
[POST]: post_type filter on index close #86

### LINKED ISSUE #86: Add in post_type filtering to the posts endpoint
We don't have any `post_type` filtering on the posts endpoint right now. We need to add that filtering in to our queries for the `index` actions.

### PR DESCRIPTION
**Pull request by snewcomer**
Added `post_type` filtering on the index route for the post controller.

### CODE REVIEW

**Comment by joshsmith** [comment]
@snewcomer would you mind adding summary here to the description?

**Comment by begedin** [comment]
~~Looks good to go~~

Completely missed the fact that we should be able to provide multiple post types in the filter. Not ready yet.

**Comment by begedin** [comment]
Should be good to go now, I changed the filter.

Could use a review by someone else.

File: test/controllers/post_controller_test.exs [hunk 1]
```diff
@@ -57,17 +57,25 @@ defmodule CodeCorps.PostControllerTest do
...
         |> json_response(200)
 
-      assert json["data"] |> Enum.count == 1
-      assert Enum.at(json["data"], 0)["attributes"]["post-type"] == post.post_type
+      assert json["data"] |> Enum.count == 2
+
+      post_types =
+        json["data"]
+        |> Enum.map(fn(r) -> r["attributes"] end)
+        |> Enum.map(fn(atts) -> atts["post-type"] end)
```
Comment (joshsmith) [inline test/controllers/post_controller_test.exs]: `r` and `atts` could be better named.

File: web/models/model_helpers.ex [hunk 2]
```diff
@@ -35,8 +35,9 @@ defmodule CodeCorps.ModelHelpers do
   end
   def organization_filter(query, _), do: query
 
-  def post_type_filter(query, %{"post_type" => post_type}) do
-    query |> where([object], object.post_type == ^post_type)
+  def post_type_filter(query, %{"post_type" => post_type_list}) do
+    post_types = post_type_list |> coalesce_string
+    query |> where([object], object.post_type in ^post_types)
```
Comment (joshsmith) [inline web/models/model_helpers.ex:40]: 👍

**Comment by begedin** [comment]
@JoshSmith Merging this, since you took a look, so I can focus on ember

**Pull request merged**